### PR TITLE
Add pipeline v1/v2 selector, add more instances

### PR DESCRIPTION
**Pull request by geyyer**
1. Add a wrapper for gridwise gemm pipelines v1 and v2

2. Refactor code using the pipelines

3. Add debug info to device-wise ops

4. Add more instances to gemm, add_add_fastgelu and batched_gemm fp16 xdl

5. Figure out what to do with failing Interwave pipeline v1 cases

6. Double check we don't break anything

### CODE REVIEW

File: include/ck/tensor_operation/gpu/device/device_batched_gemm_xdl.hpp [hunk 1]
```diff
@@ -150,7 +150,10 @@ template <typename ADataType,
           ck::index_t BBlockTransferDstScalarPerVector_K1,
           bool BBlockLdsAddExtraN,
           ck::index_t CThreadTransferSrcDstVectorDim,
-          ck::index_t CThreadTransferDstScalarPerVector>
+          ck::index_t CThreadTransferDstScalarPerVector,
+          ck::index_t NumGemmKPrefetchStage = 1,
+          ck::LoopScheduler LoopSched       = make_default_loop_scheduler(),
+          ck::index_t PipelineVersion       = 1>
```
[the remark this answers is not in the document]
Comment (asroy) [reply include/ck/tensor_operation/gpu/device/device_batched_gemm_xdl.hpp:156]: yes, that's better

[the remark this answers is not in the document]
Comment (geyyer) [reply include/ck/tensor_operation/gpu/device/device_batched_gemm_xdl.hpp:156]: Sure, something like "enum {v1, v2, ...}"?

**Comment by rosenrodt** [comment]
@geyyer This commit https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/381/commits/cbf6893363e77eac63a64314be517629a7862648 should fix issue reported in #392

File: include/ck/tensor_operation/gpu/device/device_batched_gemm_xdl.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (geyyer) [reply include/ck/tensor_operation/gpu/device/device_batched_gemm_xdl.hpp]: Fixed in [06f6f16](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/381/commits/06f6f168f9f2a3f4f47508cf29a829e4177a4054) and [adab015](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/381/commits/adab015ff22c414eaa29eccf9a6f0585c997a25e)

**Comment by geyyer** [comment]
Enable the CK_EXPERIMENTAL_INTER_WAVE_SCHEDULING, CK_EXPERIMENTAL_INTER_WAVE_INSTANCES and CK_EXPERIMENTAL_PIPELINE_V2_INSTANCES flags as we migrate to ROCm v5.3.

**Comment by zjing14** [comment]
@geyyer Please resolve conflicts.

**Comment by geyyer** [comment]
@zjing14, done with "git rebase develop". Apparently, it wasn't the best way... How do we usually rebase on top of the develop branch?

**Review by asroy — APPROVED** [review]

**Pull request merged**